### DISCUSSION TITLE
Tag naming for forked packages

### DISCUSSION THREAD

**Discussion by cweiske** [composer/composer#11380] [General]
I've forked a package and fixed a bug in it. Now I want to tag it, so that I can use the version in my project's `composer.json` file.

Let's assume the latest upstream version is v1.2.3. Shall I create a tag v1.2.4 in my own repository? This will lead to confusion when upstream releases that version, regardless if my patch has been merged or not. (because nobody can say if it's safe remove the repository from composer.json)

How can I name my tags so that it is clear that the tag is local to my forked repository?
The [official docs](https://getcomposer.org/doc/articles/versions.md#vcs-tags-and-branches) are not very clear on that, at least suffixes "alpha", "beta" and "rc" are allowed.

When trying tag names like "v1.2.4-myfork-1" composer tells me: 
> Could not parse version constraint v1.2.4-myfork-1: Invalid version string "v1.2.4-myfork-1"

**Comment by Seldaek** [comment]
I usually do not tag, and instead require the package as `dev-my-branch as 1.2.99` - see https://getcomposer.org/doc/05-repositories.md#loading-a-package-from-a-vcs-repository and https://getcomposer.org/doc/articles/aliases.md#require-inline-alias for more context.

If you do rather tag then I think I'd make it 1.2.99 or something like that, or you can use `+` to add metadata so `1.2.4+myfork` should also work

**Comment by cweiske** [comment]
Thank you! The second link is what I was looking for and did not find.

And the `+` is new to me as well.
